addons/mrp_account/models/product.py

```diff
@@ -50,7 +50,7 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves):
         dummy, bom_lines = bom.explode(self, 1)
         bom_lines = {line: data for line, data in bom_lines}
         value = 0
-        for move in stock_moves:
+        for move in stock_moves.filtered(lambda m: m.state != 'cancel'):
```

Comment:
```suggestion
        for move in stock_moves:
            if move.state == 'cancel':
                continue
```

Reply:
So we don't do 2 loops instead of 1

Reply:
@amoyaux Your wish is my command, it has been changed :smile: